etc/mpv.fish

```diff
@@ -0,0 +1,95 @@
+#
+# This file is part of mpv.
+#
+# mpv is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# mpv is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+# Add the completion for mpv. The first argument is the option with the dash
+# included and the rest of the arguments are directly passed to complete
+function _complete_mpv
+    set --local opt $argv[1]
+    set --local opt_type --short-option
+    if string match -q -r '--' -- $opt
+        set opt_type --long-option
+    end
+    complete --command mpv $opt_type (string trim --left --chars '-' -- $opt) $argv[2..]
+end
+
+function _complete_mpv_run_help
+    mpv $argv \
+        # Remove the text before the option list
+        | grep -v '^\w.*:' \
+        | string trim
+end
+
+# Print the possible values for the option
+function _complete_mpv_option -a opt
+    # We insert a tab so that it is recognized as the description of the argument
+    set --local value_description_rewrite 's/ /\t/'
+    switch $opt
+        case "--audio-device" "--vulkan-device"
+            # The value is enclosed in single quotes and the description in parentheses
+            set value_description_rewrite 's/\'([^\']+)\'\s+\((.*)\)$/\1\t\2/'
+    end
+
+    _complete_mpv_run_help $opt=help \
+        | sed -E $value_description_rewrite
+end
+
+# Print the possible values for the image option
+function _complete_mpv_image_option -a opt
+    mpv $opt=help | sed -e 's/.*: //' -e 's/ /\n/g'
+end
+
+# Print the possible values for the profile option
+function _complete_mpv_profile_option -a opt
+    _complete_mpv_run_help $opt=
+end
+
+# Cache all the mpv options
+set --local mpv_options (mpv --list-options | grep '^\s*--' | string trim)
```

Comment:
Most of the time spent when sourcing this is spent on grepping all the options here.

using fish's `string match` should be faster. I'm not too familiar with fish shell yet but using `string match --regex '^\s*--\S+'` here instead of grep reduces "time source mpv.fish" from 950ms to around 180ms. Probably could do it for both grep calls?

Reply:
Also this should use `--no-config`

Reply:
> Also this should use `--no-config`. See [15252e1](https://github.com/mpv-player/mpv/commit/15252e13d5bb367a8f86ffbc5474488b841dc96c)

Done.

Reply:
> Most of the time spent when sourcing this is spent on grepping all the options here.
> 
> using fish's `string match` should be faster. I'm not too familiar with fish shell yet but using `string match --regex '^\s*--\S+'` here instead of grep reduces "time source mpv.fish" from 950ms to around 180ms. Probably could do it for both grep calls?

I have switch it to `string match`, but on my machine I don't see such an improvement when comparing with `time source mpv.fish`.

Reply:
Weird, I also don't see that large of a difference so it's possible it was related to some sort of file caching. But it's still faster than using grep, just not by a lot.

Could you squash the commits? This looks good to me now

Reply:
> Could you squash the commits? This looks good to me now

Done.